sympy/solvers/bivariate.py

```diff
@@ -176,22 +185,73 @@ def _lambert(eq, x):
     p, den = den.as_coeff_Mul()
     e = exp(num/den)
     t = Dummy('t')
-    args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
-
     # calculating solutions from args
-    for arg in args:
+    root_list = list(roots(t**p - e, t))
+    if domain is S.Reals:
+        root_list = [r for r in root_list if not r.has(I)]
+    args = [r*d/(a*b) for r in root_list]
+    if len(args) == 0:
+        return S.EmptySet
+    # for simplicity defined two functions
+    # Lambert_Real_Braches , lambert_helper
+    def lambert_braches():
```

Comment:
It should be `lambert_branches`. The comment above refers to `Lambert_Real_Braches` which I don't think is here any more.

Reply:
oh yes...